interface/controller/app_controller.go

```diff
@@ -0,0 +1,107 @@
+package controller
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/lscantillowl/academy-go-q42021/domain/model"
+)
+
+// respondWithJSON write json response format to the response writer
+func respondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
+	response, err := json.Marshal(payload)
+	if err != nil {
+		log.Fatalf("Can not convert payload to JSON - %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	w.Write(response)
+}
+
+// Home handler to test the api
+func HomeHandler(w http.ResponseWriter, r *http.Request) {
+	resp := make(map[string]string)
+	resp["message"] = "Welcome to the Q4 GO Bootcamp API"
+	respondWithJSON(w, http.StatusOK, resp)
+}
+
+// Function to get characters from csv file and return it as json
+func GetCharacters(w http.ResponseWriter, r *http.Request) {
```

Comment:
GetCharacters and SaveCharacters should be on its own controller for characters, also keep in mind for the final delivery, you should separate this logic into three parts:

Controller: must only be in charge of calling your CharacterService and select a give it back based on error or successful result.

Service: this should be in charge of two things calling the repository and apply logic or manipulate information retrieved from Repository.

Repository: For this project it would be all the logic that connects to the csv.